docs/integrations/spark.md

```diff
@@ -523,7 +523,22 @@ Set the `fs.lakefs.*` Hadoop configurations to point to your lakeFS installation
 * `fs.lakefs.secret.key`: lakeFS secret key
 * `fs.lakefs.endpoint`: lakeFS API URL (e.g. `https://example-org.us-east-1.lakefscloud.io/api/v1`)
 
+Configure the lakeFS client to to use a temporary token instead of static credentials:
+
+* `fs.lakefs.auth.provider`: The default is `basic_auth` with `fs.lakefs.access.key` and `fs.lakefs.secret.key` for basic authentication, set to `io.lakefs.auth.TemporaryAWSCredentialsLakeFSTokenProvider` for using temporary AWS credentials, you can read more about it [here]({% link reference/security/external-principals-aws.md %}).
+
+When using `io.lakefs.auth.TemporaryAWSCredentialsLakeFSTokenProvider` as the auth provider the following configuration are relevant:
+
+* `fs.lakefs.token.aws.access.key`: AWS assumed role access key
+* `fs.lakefs.token.aws.secret.key`: AWS assumed role secret key
+* `fs.lakefs.token.aws.session.token`: AWS assumed role temporary session token
+* `fs.lakefs.token.aws.sts.endpoint`: AWS STS regional endpoint for generated the presigned-url (i.e `https://sts.us-west-2.amazonaws.com`)
+* `fs.lakefs.token.aws.sts.duration_seconds`: Optional, the duration in seconds for the initial identity token (default is 60)
+* `fs.lakefs.token.duration_seconds`: Optional, the duration in seconds for the lakeFS JWT token (default is set in the lakeFS configuration [auth.login_duration]({% link reference/configuration.md %}))
+* `fs.lakefs.token.sts.additional_headers`: Optional, comma separated list of `header:value` to attach when generating presigned sts request. Default is `X-Lakefs-Server-ID:fs.lakefs.endpoint`.
```

Comment:
I'm not sure if this is an issue or not but in the GetCallerIdentity request, these headers are separated by the ';' sign and not by a comma. It means potentially you can pass here the ';' sign as the header name or value and things will break. Or the opposite thing, users won't be able to pass a comma in header values.
Anyway, this is probably not a problem in the docs.

Reply:
That's unrelated. 
This configuration is parsed by Hadoop into lakeFS and then turned into a map for creating the pre-signed request. 
At no point this is passed "as-is" no GetCallerIdentity.